src/openloco/windows/ScenarioOptions.cpp

```diff
@@ -0,0 +1,1194 @@
+            string_ids::achieve_a_certain_performance_index,
+            string_ids::deliver_a_certain_amount_of_cargo,
+        };
+
+        // 0x0043FD51
+        static void on_dropdown(window* self, widget_index widgetIndex, int16_t itemIndex)
+        {
+            if (itemIndex == -1)
+                return;
+
+            switch (widgetIndex)
+            {
+                case widx::objective_type_btn:
+                    *objectiveType = itemIndex;
+                    self->invalidate();
+                    break;
+
+                case widx::objective_cargo_btn:
+                {
+                    uint16_t cargoIdx = dropdown::get_argument_at(itemIndex, 1);
+                    *objectiveDeliveredCargoType = cargoIdx;
+                    self->invalidate();
+                }
+            }
+        }
+
+        // 0x0043FD14
+        static void on_mouse_down(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case widx::objective_type_btn:
+                {
+                    widget_t& target = self->widgets[widx::objective_type];
+                    dropdown::show(self->x + target.left, self->y + target.top, target.width() - 3, target.height(), self->colours[1], std::size(objectiveTypeLabelIds), 0x80);
+
+                    for (size_t i = 0; i < std::size(objectiveTypeLabelIds); i++)
+                        dropdown::add(i, string_ids::dropdown_stringid, objectiveTypeLabelIds[i]);
+
+                    dropdown::set_item_selected(*objectiveType);
+                    break;
+                }
+
+                case widx::objective_value_down:
+                {
+                    switch (*objectiveType)
+                    {
+                        case scenario::objective_type::company_value:
+                            *objectiveCompanyValue = std::max<uint32_t>(*objectiveCompanyValue - 100000, scenario::min_objective_company_value);
+                            break;
+
+                        case scenario::objective_type::vehicle_profit:
+                            *objectiveMonthlyVehicleProfit = std::max<uint32_t>(*objectiveMonthlyVehicleProfit - 1000, scenario::min_objective_monthly_profit_from_vehicles);
+                            break;
+
+                        case scenario::objective_type::performance_index:
+                            *objectivePerformanceIndex = std::max<uint8_t>(*objectivePerformanceIndex - 5, scenario::min_objective_performance_index);
+                            break;
+
+                        case scenario::objective_type::cargo_delivery:
+                        {
+                            uint32_t newDeliveredCargoAmount = *objectiveDeliveredCargoAmount;
+                            newDeliveredCargoAmount -= 100;
+                            uint16_t divisor = 100;
+
+                            if (*_clickRepeatTicks >= 100)
```

Comment:
Can we change this to something that doesn't allow different branches to be reached? e.g.

```c
if (_clickRepeatTicks < 100) {
    // - 100
else if(_clickRepeatTicks <200) {
    // -1000
} else {
    //  -10000
}
```